auraed/src/cells/cell_service/cells/cell_graph.rs

```diff
@@ -0,0 +1,20 @@
+use super::CellInfo;
+
+#[derive(Clone)]
+pub struct GraphNode {
+    pub cell_info: Option<CellInfo>,
+    pub children: Vec<GraphNode>,
+}
+
+impl GraphNode {
+    pub fn with_cell_info(&self, cell_info: &CellInfo) -> GraphNode {
```

Comment:
This is really easy to use incorrectly. Instead of `&self`, the fn should take ownership of `self`.